polyfill/lib/casts.mjs

```
@@ -0,0 +1,140 @@
+  if ('string' === typeof arg) {
+    try {
+      return Time.fromString(arg);
+    } catch (ex) {}
+  }
+  if ('bigint' === typeof arg || 'number' === typeof arg || Number.isFinite(+arg))
+    return absolute(arg)
+      .getDate()
+      .inZone('UTC')
+      .getTime();
+  if ('object' === typeof arg) {
+    const { hour, minute, second, millisecond, microsecond, nanosecond } = time(arg);
+    return new Time(hour, minute, second, millisecond, microsecond, nanosecond);
+  }
+}
+
+export function yearmonth(arg) {
+  const YearMonth = ES.GetIntrinsic('%Temporal.YearMonth%');
+  if (arg instanceof YearMonth) return arg;
+  if ('string' === typeof arg) {
+    try {
+      return YearMonth.fromString(arg);
+    } catch (ex) {}
+  }
+  if ('bigint' === typeof arg || 'number' === typeof arg || Number.isFinite(+arg))
+    return absolute(arg)
+      .getDate()
+      .inZone('UTC')
+      .getYearMonth();
+  if ('object' === typeof arg) {
+    const { year, month } = arg;
+    return new YearMonth(year, month);
+  }
+}
+
+export function monthday(arg) {
+  const MonthDay = ES.GetIntrinsic('%Temporal.MonthDay%');
+  if (arg instanceof MonthDay) return arg;
+  if ('string' === typeof arg) {
+    try {
+      return MonthDay.fromString(arg);
+    } catch (ex) {}
+  }
+  if ('bigint' === typeof arg || 'number' === typeof arg || Number.isFinite(+arg))
+    return absolute(arg)
+      .getDate()
+      .inZone('UTC')
+      .getMonthDay();
+  if ('object' === typeof arg) {
+    const { month, day } = arg;
+    return new MonthDay(month, day);
+  }
+}
+
+export function duration(arg) {
+  const Duration = ES.GetIntrinsic('%Temporal.Duration%');
+  if (arg instanceof Duration) return arg;
+  if ('string' === typeof arg) {
+    try {
+      return Duration.fromString(arg);
+    } catch (ex) {}
+  }
+  if ('bigint' === typeof arg) return new Duration(0, 0, 0, 0, 0, 0, 0, 0, arg);
+  if (Number.isFinite(+arg)) return new Duration(0, 0, 0, 0, 0, 0, +arg, 0, 0);
```

I'm not sure if we should call ToNumber on all objects passed in. I think we should instead check whether its type is Number and act appropriately in that case. We could throw an error on infinities.

Separately: I am extremely skeptical of giving an overload for BigInt and Number that are semantically so different. This seems confusing. My intuition is that we should instead have fromMiliseconds and fromNanoseconds methods.

---

The duration constructor already throws on non-finite numbers. The idea is that you can pass in anything with a numeric valueOf, which seems consistent.

As to bigint vs number, I see your point. This comes from `Absolute` which returns/takes bigint for `<from|get>EpochNanoseconds()` and number from `<from|get>EpochMilliseconds()`. So the dichotomy seemed logical in that if I'm using numbers, them I'm in millisecond land and if I use bigint I'm in nanosecond land. And thanks to `+bigint` being an error the two worlds are VERY distinct.

That at least was the thought process. I don't really want to add `from<XXX>` statics on `Duration`. For that case, I'd rather just not do numeric casts, though they seem useful in old `Date` context: 
```javascript
let started = Date.now();
<do something measurable here>
let duration - Duration.from(Date.now() - started);
```

So if the options are move this to `from` statics or remove, then I'm in the remove camp.

---

#166